dockstore-webservice/src/main/java/io/dockstore/webservice/languages/NextflowHandler.java

```diff
@@ -141,6 +149,54 @@ public Version parseWorkflowContent(String filepath, String content, Set<SourceF
         return imports;
     }
 
+    /**
+     * Similar to processImports() of other handlers.
+     * The current processImports() method looks for main.nf and similar so is not suitable for recursion.
+     * Looks for imports by searching for lines that start "include"
+     * @param repositoryId  The Git repository (ex. nextflow-io/rnaseq-nf)
+     * @param content   The contents of the file being analyzed
+     * @param version   The Git version (branch/tag) of the repository
+     * @param sourceCodeRepoInterface   The source code repository interface
+     * @param workingDirectoryForFile   The parent directory of the file being analyzed
+     * @return
+     */
+    private Map<String, SourceFile> processOtherImports(String repositoryId, String content, Version version,
+            SourceCodeRepoInterface sourceCodeRepoInterface, String workingDirectoryForFile) {
+        Map<String, SourceFile> imports = new HashMap<>();
+        Pattern p = Pattern.compile("include.+?from.+?'.+?'", Pattern.DOTALL);
+        Matcher m = p.matcher(content);
+        while (m.find()) {
+            String path = getRelativeImportPathFromLine(m.group(), workingDirectoryForFile);
+            String absoluteImportPath = convertRelativePathToAbsolutePath(workingDirectoryForFile, path);
+            handleImport(repositoryId, version, imports, path, sourceCodeRepoInterface, absoluteImportPath);
+        }
+        Map<String, SourceFile> recursiveImports = new HashMap<>();
+        for (Map.Entry<String, SourceFile> importFile : imports.entrySet()) {
+            final Map<String, SourceFile> sourceFiles = processOtherImports(repositoryId, importFile.getValue().getContent(), version, sourceCodeRepoInterface, importFile.getKey());
+            recursiveImports.putAll(sourceFiles);
+        }
+        recursiveImports.putAll(imports);
+        return recursiveImports;
+    }
+
+    /**
+     * Give the line in the file that has the import, figure out what the relative path is
+     * @param line  A line in the file that has the import (ex. "include { RNASEQ } from './modules/rnaseq'")
+     * @return  The relative path
+     */
+    private static String getRelativeImportPathFromLine(String line, String workingDirectoryForFile) {
```

Comment:
I think this method is a good case for a unit test, to make sure it's catching the imports it's supposed to. Even though you already have integration tests.

It would be even better if we could use NextFlow for this, but we don't have their JAR on the classpath.

Reply:
Nextflow hasn't added support to return the imports nicely yet.

Reply:
Do like the unit test idea, if only to have a concrete example checked-in in case we change the examples on github (or is this coded for their original repos?)